pkg/modulereader/resreader.go

```diff
@@ -39,16 +41,87 @@ type VarInfo struct {
 	Required    bool
 }
 
+// OutputInfo stores information about module output values
+type OutputInfo struct {
+	Name        string
+	Description string `yaml:",omitempty"`
+	Sensitive   bool   `yaml:",omitempty"`
+	// DependsOn   []string `yaml:"depends_on,omitempty"`
+}
+
+// UnmarshalYAML supports parsing YAML OutputInfo fields as a simple list of
+// strings or as a list of maps directly into OutputInfo struct
+func (mo *OutputInfo) UnmarshalYAML(value *yaml.Node) error {
+	var name string
+
+	switch value.Kind {
```

Comment:
We could avoid this complexity (knowing types of yaml.Node) by keeping your old approach:
```
var name string
err := value.Decode(&name)
if err == nil { ... }

var fields map[string]interface{}
err = value.Decode(&fields)

if err != nil {
 return fmt.Errorf("line %d: outputs must each be a string or a map...")
}

// do fields validation.
```

Reply:
To summarize an offline discussion: there is a good argument for keeping this simple and as we learn more about custom unmarshal'ing (and limitations in `go-yaml` package), we may revert to the simple approach. For now, I've chosen to refactor the switch as an if/else that tracks both `Kind` and `Tag` to ensure that we strictly adhere to attempting to parse values that are identified as maps or strings and nothing else. If we learn that this has real world bad implications, totally happy to revert to the simple approach.

Reply:
go-yaml `Unmarshal` does not differentiate between `"0"` and `0` if recipient field in structure is `string`.
By being strict here we are going to get a behavioral disparity with rest of blueprints yaml.
Let's go easy road )

Reply:
Done. I don't like how `go-yaml` does things, but I agree we should not be unaligned with the rest of the behavior.